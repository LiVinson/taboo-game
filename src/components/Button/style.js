import styled from "styled-components"

/*
  3 types of buttons:

  button
  button primary (extension of button)
  button-lg (extension of button) or button-lg primary (tbd)

*/

export const StyledButton = styled.button`
  border: none;
  color: ${(props) => props.theme.color.secondary};
  background-color: ${(props) => props.theme.color.offWhite};
  width: 47%;
  padding: 0.6rem 0rem;
  border-radius: 2px;
  font-size: 2.3rem;
  font-family: var(--font-family--display);
  text-transform: uppercase;

  &:active {
    outline: none;
    filter: brightness(0.8);
    transform: translateY(2px);
  }
`
export const StyledPrimaryButton = styled(StyledButton)`
  color: ${(props) => props.theme.color.offWhite};
  background-color: ${(props) => props.theme.color.secondary};
  width: 47%;
  &:active {
    filter: brightness(1.2);
  }
`
