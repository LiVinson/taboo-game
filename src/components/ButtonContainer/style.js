import styled from "styled-components"


export const StyledButtonContainer = styled.div`
  margin-top: 1.5rem;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
`