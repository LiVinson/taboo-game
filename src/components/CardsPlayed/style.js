import styled from "styled-components"

export const NoCardMessage = styled.p`
	font-size:2rem;
	color: ${props => props.theme.color.grayDark2};
`