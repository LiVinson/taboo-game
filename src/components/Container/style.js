import styled from 'styled-components'

export const StyledContainer = styled.div`
	position: relative;
	width: 100vw;

	min-height: 100vh;
	overflow: hidden;
	/* 
    Used for background image. Fills entire Container div. Before used so opacity does not impact children elements */
	&:before {
		display: block;
		content: '';
		height: 100%;
		width: 100%;
		background-image: url(/img/pattern-light.png);
		opacity: 0.6;
		position: absolute;
		top: 0;
		left: 0;
	}
`
