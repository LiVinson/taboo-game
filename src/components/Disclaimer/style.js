import styled from "styled-components"

export const StyledDisclaimer = styled.p`
    margin-top: 2.5rem;
    margin-bottom: .5rem;
    font-size: 1.4rem;
    color: ${props => props.theme.color.grayDark2};
    text-align: center;
    padding: 0 .5rem;
`