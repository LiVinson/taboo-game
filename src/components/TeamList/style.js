import styled from "styled-components"
import { ReactComponent as Star } from './star-full.svg';

export const StyledTeamList = styled.div`
  &:not(:first-child) {
    margin-top: 1rem;
  }
`
export const PlayerTitle = styled.p`
  color: ${(props) => props.theme.color.grayDark2};
  font-size: 2.5rem;
  margin-bottom: 0.5rem;
  font-family: ${(props) => props.theme.text.display};

  /* Faint gray line after section title*/
  &::after {
    display: block;
    content: "";
    width: 80%;
    height: 1.5px;
    background-color: ${(props) => props.theme.color.lightGray2};
    margin: 0.8rem auto 0.6rem auto;
  }

`

export const PlayerList = styled.ul`
  color: ${(props) => props.theme.color.grayDark2};
  list-style: none;
  font-size: 2.2rem;
`
export const Player = styled.li`
  margin-bottom: 0.5rem;

`

export const StyledStar = styled(Star)`
  fill: ${(props) => props.theme.color.primary};
  height: 2rem;
  width:2rem;
`