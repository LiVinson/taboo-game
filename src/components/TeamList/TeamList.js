import React from "react"
import PropTypes from "prop-types"
import { StyledTeamList, PlayerTitle, PlayerList, Player, StyledStar } from "./style"


//players will need to be array of objects with player names and id.
//Compare playerId to currentPlayer to determine if style and icon
const TeamList = ({ title, players, currentPlayer }) => {
  return (
    <StyledTeamList>
      <PlayerTitle>{title}</PlayerTitle>
      <PlayerList>
        {players.map((player, index) => (
          // Players will see a start next to their own name
          <Player key={index}>{player.name} {currentPlayer === player.id ? <StyledStar/> : null}</Player>          

        ))}
      </PlayerList>
    </StyledTeamList>
  )
}

TeamList.propTypes = {
  title: PropTypes.string.isRequired,
  players: PropTypes.array.isRequired,
  currentPlayer: PropTypes.string.isRequired
}




export default TeamList
