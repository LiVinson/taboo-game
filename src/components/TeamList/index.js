import TeamList from "./TeamList"

export default TeamList