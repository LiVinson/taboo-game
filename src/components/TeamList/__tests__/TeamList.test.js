import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import { ThemeProvider } from 'styled-components'
import theme from '../../../global-design/theme'
import TeamList from "../TeamList"
import {Player} from "../style"


test("renders correctly", () => {
    const props = {
        title: "Team 1",
        players: [
            { id: 12345, name: "Alexa", team: "team1" },
            { id: 12346, name: "Stephen", team: "team1" },
            { id: 12348, name: "Faith", team: "team1" },
          ],
        currentPlayer: "12349"
    }
    const wrapper = renderer.create(
        <ThemeProvider theme={theme}>
            <TeamList {...props}/>
        </ThemeProvider>
        ).toJSON()

        expect(wrapper).toMatchSnapshot()
})

test("renders a Player component for each player", () => {
    const props = {
        title: "Team 1",
        players: [
            { id: 12345, name: "Alexa", team: "team1" },
            { id: 12346, name: "Stephen", team: "team1" },
            { id: 12348, name: "Faith", team: "team1" },
          ],
        currentPlayer: "12349"
    }
    const wrapper = mount(
        <ThemeProvider theme={theme}>
            <TeamList {...props}/>
        </ThemeProvider>
    )

        expect(wrapper.find(Player)).toHaveLength(props.players.length)
})
