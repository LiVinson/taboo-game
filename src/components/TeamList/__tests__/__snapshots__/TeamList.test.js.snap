// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-AxjAm fiXQlW"
>
  <p
    className="sc-AxirZ fIsTEj"
  >
    Team 1
  </p>
  <ul
    className="sc-AxiKw kWCkEn"
  >
    <li
      className="sc-AxhCb kXtvGU"
    >
      Alexa
       
    </li>
    <li
      className="sc-AxhCb kXtvGU"
    >
      Stephen
       
    </li>
    <li
      className="sc-AxhCb kXtvGU"
    >
      Faith
       
    </li>
  </ul>
</div>
`;
